beater/beater.go

```diff
@@ -308,6 +308,11 @@ func newServerRunner(ctx context.Context, args serverRunnerParams) (*serverRunne
 	if cfg.DataStreams.Enabled && args.KibanaConfig != nil {
 		cfg.Kibana.ClientConfig = *args.KibanaConfig
 	}
+	if args.Beat.Manager != nil && args.Beat.Manager.Enabled() {
+		// If we're running in managed mode we do not want to
+		// communicate with kibana.
+		cfg.Kibana.Enabled = false
+	}
```

Comment:
For some reason I can't find the code again, but our default unmarshal behavior for bool type is to set it to true if it's not found. Even if it is set, apm-server doesn't communicate with kibana if running in managed mode.

Reply:
Are you thinking of the [`libbeat/common.Config.Enabled()`](https://github.com/elastic/beats/blob/27e76c56771103326a62a9e46a58f345ed7eda21/libbeat/common/config.go#L280-L294) method, which returns true when the config block is defined but the "enabled" field is not explicitly specified?

I'm not sure if that's relevant here, since we only set `cfg.Kibana.ClientConfig` above.

Reply:
Ah, I see the problem. I just created a default apm policy and I get this:

![image](https://user-images.githubusercontent.com/843579/120254233-f16c1f80-c2bb-11eb-9315-c6a4d3c569ed.png)

The presence of that `kibana: ...` means it is enabled. This config was introduced as a temporary measure to support central config via Kibana; we should remove it in 7.14:

https://github.com/elastic/apm-server/blob/16618f2a00400dcde109c00037be79e24f3938c5/apmpackage/apm/0.2.0/manifest.yml#L62-L67

https://github.com/elastic/apm-server/blob/16618f2a00400dcde109c00037be79e24f3938c5/apmpackage/apm/0.2.0/agent/input/template.yml.hbs#L7-L8

Reply:
can we remove this change now?

Reply:
removed